2025-08-19 09:27:54,644 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Tea Bags - English Breakfast (Twinings) (1)
2025-08-19 09:28:10,685 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:619) - Item Tea Bags - English Breakfast (Twinings) addedd successfully.
2025-08-19 09:28:10,686 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Torani - Caramel Sauce (2)
2025-08-19 09:28:11,748 - WARNING - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:565) - Failed to add item Torani - Caramel Sauce: Message: element click intercepted: Element <a tabindex="" class="btn btn-outline-primary btn-sm" data-ember-action="" data-ember-action-111="111">...</a> is not clickable at point (378, 970). Other element would receive the click: <div class="modal-backdrop fade"></div>
  (Session info: chrome=139.0.7258.128)
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd62be0]
	(No symbol) [0x0xd60f9a]
	(No symbol) [0x0xd5eaf7]
	(No symbol) [0x0xd5ddb4]
	(No symbol) [0x0xd52565]
	(No symbol) [0x0xd813dc]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 563, in _input_transfer_items
    self._add_transfer_item(item)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 578, in _add_transfer_item
    add_item_button.click()
    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 119, in click
    self._execute(Command.CLICK_ELEMENT)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 570, in _execute
    return self._parent.execute(command, params)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <a tabindex="" class="btn btn-outline-primary btn-sm" data-ember-action="" data-ember-action-111="111">...</a> is not clickable at point (378, 970). Other element would receive the click: <div class="modal-backdrop fade"></div>
  (Session info: chrome=139.0.7258.128)
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd62be0]
	(No symbol) [0x0xd60f9a]
	(No symbol) [0x0xd5eaf7]
	(No symbol) [0x0xd5ddb4]
	(No symbol) [0x0xd52565]
	(No symbol) [0x0xd813dc]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 09:28:11,750 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Torani - Dark Chocolate Sauce (2)
2025-08-19 09:28:28,180 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:619) - Item Torani - Dark Chocolate Sauce addedd successfully.
2025-08-19 09:28:28,180 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Torani - White Chocolate Sauce (1)
2025-08-19 09:28:29,298 - WARNING - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:565) - Failed to add item Torani - White Chocolate Sauce: Message: element click intercepted: Element <a tabindex="" class="btn btn-outline-primary btn-sm" data-ember-action="" data-ember-action-111="111">...</a> is not clickable at point (378, 970). Other element would receive the click: <div class="modal-backdrop fade"></div>
  (Session info: chrome=139.0.7258.128)
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd62be0]
	(No symbol) [0x0xd60f9a]
	(No symbol) [0x0xd5eaf7]
	(No symbol) [0x0xd5ddb4]
	(No symbol) [0x0xd52565]
	(No symbol) [0x0xd813dc]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 563, in _input_transfer_items
    self._add_transfer_item(item)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 578, in _add_transfer_item
    add_item_button.click()
    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 119, in click
    self._execute(Command.CLICK_ELEMENT)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 570, in _execute
    return self._parent.execute(command, params)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <a tabindex="" class="btn btn-outline-primary btn-sm" data-ember-action="" data-ember-action-111="111">...</a> is not clickable at point (378, 970). Other element would receive the click: <div class="modal-backdrop fade"></div>
  (Session info: chrome=139.0.7258.128)
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd62be0]
	(No symbol) [0x0xd60f9a]
	(No symbol) [0x0xd5eaf7]
	(No symbol) [0x0xd5ddb4]
	(No symbol) [0x0xd52565]
	(No symbol) [0x0xd813dc]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 09:28:29,300 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Whitefish Salad (1)
2025-08-19 09:28:45,127 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:619) - Item Whitefish Salad addedd successfully.
2025-08-19 09:28:45,127 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Wrap Foil - 9x10.75 (1)
2025-08-19 09:28:46,191 - WARNING - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:565) - Failed to add item Wrap Foil - 9x10.75: Message: element click intercepted: Element <a tabindex="" class="btn btn-outline-primary btn-sm" data-ember-action="" data-ember-action-111="111">...</a> is not clickable at point (378, 970). Other element would receive the click: <div class="modal-backdrop fade"></div>
  (Session info: chrome=139.0.7258.128)
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd62be0]
	(No symbol) [0x0xd60f9a]
	(No symbol) [0x0xd5eaf7]
	(No symbol) [0x0xd5ddb4]
	(No symbol) [0x0xd52565]
	(No symbol) [0x0xd813dc]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 563, in _input_transfer_items
    self._add_transfer_item(item)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 578, in _add_transfer_item
    add_item_button.click()
    ~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 119, in click
    self._execute(Command.CLICK_ELEMENT)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 570, in _execute
    return self._parent.execute(command, params)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element <a tabindex="" class="btn btn-outline-primary btn-sm" data-ember-action="" data-ember-action-111="111">...</a> is not clickable at point (378, 970). Other element would receive the click: <div class="modal-backdrop fade"></div>
  (Session info: chrome=139.0.7258.128)
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd62be0]
	(No symbol) [0x0xd60f9a]
	(No symbol) [0x0xd5eaf7]
	(No symbol) [0x0xd5ddb4]
	(No symbol) [0x0xd52565]
	(No symbol) [0x0xd813dc]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 09:28:46,193 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:562) - Adding transfer item: Wrap Paper - 15x10.75 (2)
2025-08-19 09:28:54,845 - WARNING - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:565) - Failed to add item Wrap Paper - 15x10.75: list index out of range
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 563, in _input_transfer_items
    self._add_transfer_item(item)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 603, in _add_transfer_item
    item_choices[0].click()
    ~~~~~~~~~~~~^^^
IndexError: list index out of range
2025-08-19 09:28:54,846 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:567) - All items processed.
2025-08-19 09:28:54,846 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:467) - Transfer for 45 items from Bakery to Easthill on 2025-08-16 completed successfully.
2025-08-19 09:28:54,847 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:452) - Starting transfer protocol for 32 items from Bakery to Triphammer on 2025-08-15
2025-08-19 09:29:03,638 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:473) - Creating new transfer in Craftable.
2025-08-19 09:29:07,704 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:483) - Entering transfer details: Bakery to Triphammer on 2025-08-15
2025-08-19 09:29:07,723 - ERROR - backend.bots.craftable_bot.craftable_bot.CraftableBot - (logger.py:107) - Error in input_transfer: Message: no such element: Unable to locate element: {"method":"xpath","selector":".//form"}
  (Session info: chrome=139.0.7258.128); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd5c89e]
	(No symbol) [0x0xd5cc3b]
	(No symbol) [0x0xd520d1]
	(No symbol) [0x0xd81424]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\utils\logger.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 463, in input_transfer
    self._enter_transfer_details(transfer)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 487, in _enter_transfer_details
    transfer_form = transfer_modal.find_element(By.XPATH, './/form')
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 599, in find_element
    return self._execute(Command.FIND_CHILD_ELEMENT, {"using": by, "value": value})["value"]
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 570, in _execute
    return self._parent.execute(command, params)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"xpath","selector":".//form"}
  (Session info: chrome=139.0.7258.128); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd5c89e]
	(No symbol) [0x0xd5cc3b]
	(No symbol) [0x0xd520d1]
	(No symbol) [0x0xd81424]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 09:29:07,725 - ERROR - backend.bots.craftable_bot.craftable_bot.CraftableBot - (logger.py:107) - Error in input_transfers: Message: no such element: Unable to locate element: {"method":"xpath","selector":".//form"}
  (Session info: chrome=139.0.7258.128); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd5c89e]
	(No symbol) [0x0xd5cc3b]
	(No symbol) [0x0xd520d1]
	(No symbol) [0x0xd81424]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\utils\logger.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 444, in input_transfers
    self.input_transfer(transfer)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\utils\logger.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 463, in input_transfer
    self._enter_transfer_details(transfer)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 487, in _enter_transfer_details
    transfer_form = transfer_modal.find_element(By.XPATH, './/form')
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 599, in find_element
    return self._execute(Command.FIND_CHILD_ELEMENT, {"using": by, "value": value})["value"]
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webelement.py", line 570, in _execute
    return self._parent.execute(command, params)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"xpath","selector":".//form"}
  (Session info: chrome=139.0.7258.128); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Stacktrace:
	GetHandleVerifier [0x0xedfd33+62915]
	GetHandleVerifier [0x0xedfd74+62980]
	(No symbol) [0x0xd13e13]
	(No symbol) [0x0xd5c89e]
	(No symbol) [0x0xd5cc3b]
	(No symbol) [0x0xd520d1]
	(No symbol) [0x0xd81424]
	(No symbol) [0x0xd51ff4]
	(No symbol) [0x0xd81654]
	(No symbol) [0x0xda26ea]
	(No symbol) [0x0xd811d6]
	(No symbol) [0x0xd50833]
	(No symbol) [0x0xd516a4]
	GetHandleVerifier [0x0x1148d23+2590131]
	GetHandleVerifier [0x0x1143f6a+2570234]
	GetHandleVerifier [0x0xf059ea+217722]
	GetHandleVerifier [0x0xef6058+153832]
	GetHandleVerifier [0x0xefc4bd+179533]
	GetHandleVerifier [0x0xee7738+94152]
	GetHandleVerifier [0x0xee78c2+94546]
	GetHandleVerifier [0x0xed2bda+9322]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 11:28:33,198 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:168) - Closing WebDriver session.
2025-08-19 11:28:33,268 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:175) - WebDriver session closed.
2025-08-19 11:28:48,560 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:131) - Logging into Craftable.
2025-08-19 11:29:01,996 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:155) - Login successful.
2025-08-19 11:29:01,996 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:254) - Starting order download protocol.
2025-08-19 11:29:01,997 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:257) - Accessing order page for Bakery.
2025-08-19 11:29:08,629 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for UNFI (08/16/2025).
2025-08-19 11:29:10,982 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / UNFI / 20250816.
2025-08-19 11:29:11,001 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for UNFI.
2025-08-19 11:29:15,477 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Sysco (08/16/2025).
2025-08-19 11:29:18,250 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Sysco / 20250816.
2025-08-19 11:29:18,273 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Sysco.
2025-08-19 11:29:22,753 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for FingerLakes Farms (08/16/2025).
2025-08-19 11:29:25,168 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / FingerLakes Farms / 20250816.
2025-08-19 11:29:25,205 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for FingerLakes Farms.
2025-08-19 11:29:29,692 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Hillcrest Dairy (08/16/2025).
2025-08-19 11:29:31,119 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Hillcrest Dairy / 20250816.
2025-08-19 11:29:31,139 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Hillcrest Dairy.
2025-08-19 11:29:35,558 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Performance Food (08/16/2025).
2025-08-19 11:29:38,623 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Performance Food / 20250816.
2025-08-19 11:29:38,646 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Performance Food.
2025-08-19 11:29:43,089 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for US Foods (08/17/2025).
2025-08-19 11:29:45,805 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / US Foods / 20250817.
2025-08-19 11:29:45,827 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for US Foods.
2025-08-19 11:29:50,276 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Hillcrest Foods (08/17/2025).
2025-08-19 11:29:52,678 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Hillcrest Foods / 20250817.
2025-08-19 11:29:52,697 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Hillcrest Foods.
2025-08-19 11:29:57,151 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Keck's Food Service (08/17/2025).
2025-08-19 11:29:58,673 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Keck's Food Service / 20250817.
2025-08-19 11:29:58,693 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Keck's Food Service.
2025-08-19 11:30:03,115 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Webstaurant (08/17/2025).
2025-08-19 11:30:05,266 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Webstaurant / 20250817.
2025-08-19 11:30:05,288 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Webstaurant.
2025-08-19 11:30:09,715 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Eurocafe Imports (08/17/2025).
2025-08-19 11:30:10,634 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Eurocafe Imports / 20250817.
2025-08-19 11:30:10,654 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Eurocafe Imports.
2025-08-19 11:30:15,083 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Copper Horse Coffee (08/17/2025).
2025-08-19 11:30:16,444 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Copper Horse Coffee / 20250817.
2025-08-19 11:30:16,462 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Copper Horse Coffee.
2025-08-19 11:30:20,886 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Equal Exchange (08/17/2025).
2025-08-19 11:30:22,349 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Equal Exchange / 20250817.
2025-08-19 11:30:22,369 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Equal Exchange.
2025-08-19 11:30:26,783 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Coca-Cola (08/17/2025).
2025-08-19 11:30:28,584 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Coca-Cola / 20250817.
2025-08-19 11:30:28,604 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Coca-Cola.
2025-08-19 11:30:33,022 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Regional Distributors, Inc. (08/18/2025).
2025-08-19 11:30:35,534 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Regional Distributors, Inc. / 20250818.
2025-08-19 11:30:35,554 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Regional Distributors, Inc..
2025-08-19 11:30:39,989 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Lentz (08/18/2025).
2025-08-19 11:30:41,568 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Lentz / 20250818.
2025-08-19 11:30:41,591 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Lentz.
2025-08-19 11:30:46,023 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for BALKAN BEVERAGE LLC (08/18/2025).
2025-08-19 11:30:47,438 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / BALKAN BEVERAGE LLC / 20250818.
2025-08-19 11:30:47,458 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for BALKAN BEVERAGE LLC.
2025-08-19 11:30:51,878 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for A.L. George (08/18/2025).
2025-08-19 11:30:53,418 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / A.L. George / 20250818.
2025-08-19 11:30:53,436 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for A.L. George.
2025-08-19 11:30:57,867 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Dawn (08/18/2025).
2025-08-19 11:30:59,296 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Dawn / 20250818.
2025-08-19 11:30:59,313 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Dawn.
2025-08-19 11:31:03,729 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for DUTCH VALLEY FOOD DIST (08/18/2025).
2025-08-19 11:31:04,948 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / DUTCH VALLEY FOOD DIST / 20250818.
2025-08-19 11:31:04,964 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for DUTCH VALLEY FOOD DIST.
2025-08-19 11:31:09,375 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Side Hill Acres (08/18/2025).
2025-08-19 11:31:10,535 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Side Hill Acres / 20250818.
2025-08-19 11:31:10,553 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Side Hill Acres.
2025-08-19 11:31:14,973 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Johnston Paper (08/19/2025).
2025-08-19 11:31:16,489 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Johnston Paper / 20250819.
2025-08-19 11:31:16,508 - ERROR - backend.bots.craftable_bot.craftable_bot.CraftableBot - (logger.py:107) - Error in download_orders: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\andrew\\Desktop\\Andrew\\Projects\\IthacaBakery\\RandomStuff\\WorkBot\\refactor\\data\\orders\\Johnston Paper\\Johnston Paper_Bakery_2025-08-19.pdf'
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\utils\logger.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 269, in download_orders
    self._process_order_row(store, stale_reference_table_rows[pos], vendors, download_pdf, update)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 306, in _process_order_row
    if update and self._update_existing_order(order_to_check_update):
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 704, in _update_existing_order
    return self.order_coordinator.check_and_update_order(order)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\coordinators\order_coordinator.py", line 210, in check_and_update_order
    self.file_handler.remove_file(file_path.with_suffix('.pdf'))
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\storage\file\file_handler.py", line 60, in remove_file
    path.unlink()
    ~~~~~~~~~~~^^
  File "C:\Users\Will\AppData\Local\Programs\Python\Python313\Lib\pathlib\_local.py", line 746, in unlink
    os.unlink(self)
    ~~~~~~~~~^^^^^^
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\andrew\\Desktop\\Andrew\\Projects\\IthacaBakery\\RandomStuff\\WorkBot\\refactor\\data\\orders\\Johnston Paper\\Johnston Paper_Bakery_2025-08-19.pdf'
2025-08-19 11:31:52,985 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:254) - Starting order download protocol.
2025-08-19 11:31:52,985 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:257) - Accessing order page for Bakery.
2025-08-19 11:32:00,167 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Johnston Paper (08/19/2025).
2025-08-19 11:32:01,621 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Johnston Paper / 20250819.
2025-08-19 11:32:01,621 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Johnston Paper.
2025-08-19 11:32:06,120 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:271) - Order download complete.
2025-08-19 11:32:06,121 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:272) - Closing WebDriver session.
2025-08-19 11:32:50,642 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:254) - Starting order download protocol.
2025-08-19 11:32:50,642 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:257) - Accessing order page for Triphammer.
2025-08-19 11:32:50,643 - ERROR - backend.bots.craftable_bot.craftable_bot.CraftableBot - (logger.py:107) - Error in download_orders: Message: invalid session id
Stacktrace:
	GetHandleVerifier [0x0xe5fd33+62915]
	GetHandleVerifier [0x0xe5fd74+62980]
	(No symbol) [0x0xc93c70]
	(No symbol) [0x0xccfaf8]
	(No symbol) [0x0xd01296]
	(No symbol) [0x0xcfce75]
	(No symbol) [0x0xcfc3f6]
	(No symbol) [0x0xc65755]
	(No symbol) [0x0xc65cae]
	(No symbol) [0x0xc6613d]
	GetHandleVerifier [0x0x10c8d23+2590131]
	GetHandleVerifier [0x0x10c3f6a+2570234]
	GetHandleVerifier [0x0xe859ea+217722]
	GetHandleVerifier [0x0xe76058+153832]
	GetHandleVerifier [0x0xe7c4bd+179533]
	(No symbol) [0x0xc65420]
	(No symbol) [0x0xc64c36]
	GetHandleVerifier [0x0x120509c+3885356]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\utils\logger.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 259, in download_orders
    self.driver.get(store_order_url)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\undetected_chromedriver\__init__.py", line 665, in get
    return super().get(url)
           ~~~~~~~~~~~^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 454, in get
    self.execute(Command.GET, {"url": url})
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.InvalidSessionIdException: Message: invalid session id
Stacktrace:
	GetHandleVerifier [0x0xe5fd33+62915]
	GetHandleVerifier [0x0xe5fd74+62980]
	(No symbol) [0x0xc93c70]
	(No symbol) [0x0xccfaf8]
	(No symbol) [0x0xd01296]
	(No symbol) [0x0xcfce75]
	(No symbol) [0x0xcfc3f6]
	(No symbol) [0x0xc65755]
	(No symbol) [0x0xc65cae]
	(No symbol) [0x0xc6613d]
	GetHandleVerifier [0x0x10c8d23+2590131]
	GetHandleVerifier [0x0x10c3f6a+2570234]
	GetHandleVerifier [0x0xe859ea+217722]
	GetHandleVerifier [0x0xe76058+153832]
	GetHandleVerifier [0x0xe7c4bd+179533]
	(No symbol) [0x0xc65420]
	(No symbol) [0x0xc64c36]
	GetHandleVerifier [0x0x120509c+3885356]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 11:43:47,690 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:168) - Closing WebDriver session.
2025-08-19 11:43:47,692 - ERROR - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:173) - Error closing WebDriver session: Message: invalid session id
Stacktrace:
	GetHandleVerifier [0x0xe5fd33+62915]
	GetHandleVerifier [0x0xe5fd74+62980]
	(No symbol) [0x0xc93c70]
	(No symbol) [0x0xccfaf8]
	(No symbol) [0x0xd01296]
	(No symbol) [0x0xcfce75]
	(No symbol) [0x0xcfc3f6]
	(No symbol) [0x0xc65755]
	(No symbol) [0x0xc65cae]
	(No symbol) [0x0xc6613d]
	GetHandleVerifier [0x0x10c8d23+2590131]
	GetHandleVerifier [0x0x10c3f6a+2570234]
	GetHandleVerifier [0x0xe859ea+217722]
	GetHandleVerifier [0x0xe76058+153832]
	GetHandleVerifier [0x0xe7c4bd+179533]
	(No symbol) [0x0xc65420]
	(No symbol) [0x0xc64c36]
	GetHandleVerifier [0x0x120509c+3885356]
	BaseThreadInitThunk [0x0x755dfcc9+25]
	RtlGetAppContainerNamedObjectPath [0x0x776d82ae+286]
	RtlGetAppContainerNamedObjectPath [0x0x776d827e+238]

2025-08-19 11:43:47,692 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:175) - WebDriver session closed.
2025-08-19 11:44:01,917 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:131) - Logging into Craftable.
2025-08-19 11:44:09,164 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:155) - Login successful.
2025-08-19 11:44:09,164 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:254) - Starting order download protocol.
2025-08-19 11:44:09,164 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:257) - Accessing order page for Triphammer.
2025-08-19 11:44:15,795 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Sysco (08/15/2025).
2025-08-19 11:44:19,054 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Triphammer / Sysco / 20250815.
2025-08-19 11:44:19,077 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for Sysco.
2025-08-19 11:44:24,253 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:271) - Order download complete.
2025-08-19 11:44:24,254 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:272) - Closing WebDriver session.
2025-08-19 11:54:37,155 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:131) - Logging into Craftable.
2025-08-19 11:54:44,472 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:155) - Login successful.
2025-08-19 11:54:44,472 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:254) - Starting order download protocol.
2025-08-19 11:54:44,473 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:257) - Accessing order page for Triphammer.
2025-08-19 11:54:51,370 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for US Foods (08/15/2025).
2025-08-19 11:54:54,365 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Triphammer / US Foods / 20250815.
2025-08-19 11:54:54,386 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:312) - Saving order for US Foods.
2025-08-19 11:54:59,289 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:271) - Order download complete.
2025-08-19 11:54:59,290 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:272) - Closing WebDriver session.
2025-08-19 12:42:50,099 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:168) - Closing WebDriver session.
2025-08-19 12:42:55,254 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:175) - WebDriver session closed.
2025-08-19 12:44:16,072 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:131) - Logging into Craftable.
2025-08-19 12:44:23,290 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:155) - Login successful.
2025-08-19 12:44:23,290 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:254) - Starting order download protocol.
2025-08-19 12:44:23,290 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:257) - Accessing order page for Bakery.
2025-08-19 12:44:30,212 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:298) - Retrieving order for Webstaurant (08/17/2025).
2025-08-19 12:44:32,335 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:703) - [Order Update] Initiating update protocol for Bakery / Webstaurant / 20250817.
2025-08-19 12:44:32,355 - ERROR - backend.bots.craftable_bot.craftable_bot.CraftableBot - (logger.py:107) - Error in download_orders: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\andrew\\Desktop\\Andrew\\Projects\\IthacaBakery\\RandomStuff\\WorkBot\\refactor\\data\\orders\\Webstaurant\\Webstaurant_Bakery_2025-08-17.pdf'
Traceback (most recent call last):
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\utils\logger.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 269, in download_orders
    self._process_order_row(store, stale_reference_table_rows[pos], vendors, download_pdf, update)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 306, in _process_order_row
    if update and self._update_existing_order(order_to_check_update):
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\bots\craftable_bot\craftable_bot.py", line 704, in _update_existing_order
    return self.order_coordinator.check_and_update_order(order)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\coordinators\order_coordinator.py", line 210, in check_and_update_order
    self.file_handler.remove_file(file_path.with_suffix('.pdf'))
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andrew\Desktop\Andrew\Projects\IthacaBakery\RandomStuff\WorkBot\refactor\backend\storage\file\file_handler.py", line 60, in remove_file
    path.unlink()
    ~~~~~~~~~~~^^
  File "C:\Users\Will\AppData\Local\Programs\Python\Python313\Lib\pathlib\_local.py", line 746, in unlink
    os.unlink(self)
    ~~~~~~~~~^^^^^^
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\andrew\\Desktop\\Andrew\\Projects\\IthacaBakery\\RandomStuff\\WorkBot\\refactor\\data\\orders\\Webstaurant\\Webstaurant_Bakery_2025-08-17.pdf'
2025-08-19 12:48:57,351 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:168) - Closing WebDriver session.
2025-08-19 12:49:02,350 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:175) - WebDriver session closed.
2025-08-19 13:30:59,847 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:168) - Closing WebDriver session.
2025-08-19 13:31:05,826 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:175) - WebDriver session closed.
2025-08-19 14:31:23,093 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:168) - Closing WebDriver session.
2025-08-19 14:31:28,537 - INFO - backend.bots.craftable_bot.craftable_bot.CraftableBot - (craftable_bot.py:175) - WebDriver session closed.
